packages/permission-log-controller/tests/PermissionLogController.test.ts

```diff
@@ -366,97 +368,101 @@ describe('PermissionLogController', () => {
         expect(entry2).toStrictEqual(controller.state.permissionActivityLog[1]);
       });
 
-      it('ignores expected methods', () => {
+      it('ignores activity for expected methods', () => {
+        const controller = initController();
+        const logMiddleware = initMiddleware(controller);
         expect(controller.state.permissionActivityLog).toHaveLength(0);
 
         const res = {
           ...PendingJsonRpcResponseStruct.TYPE,
           result: ['bar'],
         };
 
-        logMiddleware(
+        const ignoredMethods = [
           RPC_REQUESTS.metamask_sendDomainMetadata(SUBJECTS.c.origin, 'foobar'),
-          res,
-          mockNext,
-          noop,
-        );
-        logMiddleware(
           RPC_REQUESTS.custom(SUBJECTS.b.origin, 'eth_getBlockNumber'),
-          res,
-          mockNext,
-          noop,
-        );
-        logMiddleware(
           RPC_REQUESTS.custom(SUBJECTS.b.origin, 'net_version'),
-          res,
-          mockNext,
-          noop,
-        );
+        ];
+
+        ignoredMethods.forEach((req) => {
+          logMiddleware(req, res, mockNext, noop);
+        });
 
         expect(controller.state.permissionActivityLog).toHaveLength(0);
       });
 
-      it('enforces log limit', () => {
-        const req = RPC_REQUESTS.test_method(SUBJECTS.a.origin);
-        const res = {
-          ...PendingJsonRpcResponseStruct.TYPE,
-          result: ['bar'],
-        };
+      describe('log limit enforcement', () => {
```

Comment:
In the [unit testing contributor guide](https://github.com/MetaMask/contributor-docs/blob/main/docs/unit-testing.md#-use-describe-to-group-tests-under-scenarios) I recommended that we use `describe` not to group together tests by behavior (what the thing we're testing does), but rather by scenario (_why_ the thing we're testing is doing what it's doing). I know different people have different ways of structuring tests, so there's not necessarily one right answer here. However, I feel that grouping tests in this way makes it easier to compare tests to implementation to ensure that you have complete coverage. The only real way to know whether you have complete coverage for a piece of code is to map out all of the code paths (all of the routes that the code will take through different branches of logic). When mapping out the code paths you end up producing a list that looks something like this:

- When X is true and Y is true: do something
- When X is true and Y is false: do something else
- When X is false: do yet another thing

So, structuring tests by using `describe` to describe the context that creates the behavior (the part before the ":" here) and then using `it` to describe the behavior (the part afterward) is a natural outcome of this exercise:

``` typescript
describe('when X is true and Y is true', () => {
  it('does something', () => {
    // ...
  });
});

describe('when X is true and Y is false', () => {
  it('does something else', () => {
    // ...
  });
});

describe('when X is false', () => {
  it('does yet another thing', () => {
    // ...
  });
});
```

Considering the way that these tests are presently written, using this structure throughout this file would result in rewriting it completely. However, perhaps it's worth it to nudge them in that direction? In other words, maybe you don't need this one `describe` in particular?

What are your thoughts?